### PR TITLE
NFC - Eliminate query_for_ids

### PR DESCRIPTION
**Pull request by fluidvanadium**
lives on #1239

### CODE REVIEW

File: libtonode-tests/tests/concrete.rs [hunk 1]
```diff
@@ -4330,7 +4330,26 @@ async fn propose_orchard_dust_to_sapling() {
...
+            600_000,
+            Some("600_000 orchard funds"),
+        )],
+    )
+    .await
+    .unwrap();
+    increase_height_and_wait_for_client(&regtest_manager, &recipient, 1)
+        .await
+        .unwrap();
+    let lapo = dbg!(recipient.list_anypool_outputs().await);
```
Comment (zancas) [inline libtonode-tests/tests/concrete.rs]: dbg

File: zingolib/src/lightclient/describe.rs [hunk 2]
```diff
@@ -800,12 +786,28 @@ impl LightClient {
...
+    pub async fn list_anypool_outputs(&self) -> Vec<crate::wallet::notes::AnyPoolOutput> {
+        self.wallet
+            .transaction_context
+            .transaction_metadata_set
+            .read()
+            .await
+            .transaction_records_by_id
+            .0
+            .values()
+            .flat_map(|record| record.get_all_outputs())
```
Comment (zancas) [inline zingolib/src/lightclient/describe.rs]: I think we want to use AnyPoolOuts as the higher layer interface...  maybe soonish deleting/rewriting TransactionRecord.

File: zingolib/src/wallet/notes.rs [hunk 3]
```diff
@@ -30,55 +31,35 @@ pub enum AnyPoolOutput {
...
-                    .map(|output| Self::SaplingNote(output.clone())),
-            )
-            .chain(
-                transaction_record
-                    .orchard_notes
-                    .iter()
-                    .map(|output| Self::OrchardNote(output.clone())),
-            )
+    /// this function filters a vec of outputs by pool
+    pub fn filter_outputs_pools(list: Vec<Self>, filter: OutputPoolQuery) -> Vec<Self> {
```
Comment (zancas) [inline zingolib/src/wallet/notes.rs]: Why get outputs from a specific pool, if the type is itself named "AnyPool"?

Comment (zancas) [reply zingolib/src/wallet/notes.rs]: Are there callers of this method yet?

File: zingolib/src/lightclient/describe.rs [hunk 4]
```diff
@@ -800,12 +786,28 @@ impl LightClient {
...
+    pub async fn list_outputs(&self) -> Vec<crate::wallet::notes::AnyPoolOutput> {
+        self.wallet
+            .transaction_context
+            .transaction_metadata_set
+            .read()
+            .await
+            .transaction_records_by_id
+            .0
+            .values()
+            .flat_map(|record| record.get_outputs())
```
Comment (zancas) [inline zingolib/src/lightclient/describe.rs]: I prefer `AnyPoolOutputs::get_outputs(record)`.   This unifies the interface through `AnyPoolOutputs`.   Then later in life we can have a `ShieldedOutputs`..   etc..

Comment (fluidvanadium) [reply zingolib/src/lightclient/describe.rs]: hmm

Comment (fluidvanadium) [reply zingolib/src/lightclient/describe.rs]: i acknowledge this direction

Comment (fluidvanadium) [reply zingolib/src/lightclient/describe.rs]: Seems like having a ShieldedOutputs enum could reduce compiler uncertainty in some cases. On the other hand, its futureproofing which may not even pan out.

Comment (fluidvanadium) [reply zingolib/src/lightclient/describe.rs]: AnyPoolOutputs::from_transaction_record(record) could be a precise pattern

**Comment by zancas** [comment]
I fixed the merge conflicts for this.

**Review by zancas — APPROVED** [review]

**Review by zancas — APPROVED** [review]

**Pull request merged**